File: 2023-2024/sourcecode/mod1a/lab2/42DynamicFragmentBasicsJava/app/src/main/java/com/example/a42_dynamicfragmentbasics_java/FragmentA.java
package com.example.a42_dynamicfragmentbasics_java;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.a42_dynamicfragmentbasics_java.databinding.FragmentABinding;

public class FragmentA extends Fragment {

    private FragmentABinding binding;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentABinding.inflate(inflater);

        binding.startFragmentB
                .setOnClickListener(v -> requireActivity().getSupportFragmentManager()
                        .beginTransaction()
                        .replace(R.id.fragment_container_view_tag, new FragmentB())
                        .addToBackStack(null)
                        .commit());

        return binding.getRoot();
    }
}

File: 2023-2024/sourcecode/mod1a/lab2/41DynamicFragmentBasicsKotlin/app/src/main/java/com/example/a41_dynamicfragmentbasics_kotlin/FragmentA.kt
package com.example.a41_dynamicfragmentbasics_kotlin

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.a41_dynamicfragmentbasics_kotlin.databinding.FragmentABinding

class FragmentA : Fragment() {

    private lateinit var binding: FragmentABinding

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentABinding.inflate(inflater)
        binding.startFragmentB.setOnClickListener {
            requireActivity().supportFragmentManager
                .beginTransaction()
                .replace(R.id.fragment_container_view_tag, FragmentB())
                .addToBackStack(null)
                .commit()
        }
        return binding.root
    }
}